### PR TITLE
New parameter 'ids' on document get endpoint to retrieve multiple documents by id at once

### LINKED ISSUE #1226: A way to get several documents at once by submitting a list of ids
Currently the document interface takes an id field, but it would be convenient to be able to supply a list of ids instead.

### PR DESCRIPTION
**Pull request by matko**
Fixes #1226.

### CODE REVIEW

**Comment by matko** [comment]
I rewrote the document endpoint so that getting by a single id is now a special case of getting by several ids. I then created new parameters in both the cli tool and the server endpoint to use this with multiple ids (supporting both as a query param or inside of a submitted query document). If a single id is submitted, this will take precedence over any id list submitted.

**Comment by hoijnet** [comment]
This is so beautiful!!!

File: src/server/routes.pl [hunk 1]
```diff
@@ -497,9 +505,9 @@
 
             api_read_document_selector(
                 System_DB, Auth, Path, Graph_Type,
-                Id, Type, Query, Config,
+                Id, Ids, Type, Query, Config,
                 Requested_Data_Version, Actual_Data_Version,
-                cors_json_stream_write_headers_(Request, Actual_Data_Version)
+                routes:cors_json_stream_write_headers_(Request, Actual_Data_Version)
```
Comment (rrooij) [inline src/server/routes.pl:510]: ??

File: src/cli/main.pl [hunk 2]
```diff
@@ -1077,6 +1077,12 @@
            shortflags([i]),
            default('_'),
            help('id of document to retrieve')],
+          [opt(ids),
+           % todo something something comma separated values
```
Comment (rrooij) [inline src/cli/main.pl:1081]: ? :O

**Review by rrooij — APPROVED** [review]

File: src/server/routes.pl (hunk 1 above)
Comment (matko) [reply src/server/routes.pl:510]: yeah i don't know. it started complaining here. slightly cargo cult but what can you do.

File: src/cli/main.pl (hunk 2 above)
Comment (matko) [reply src/cli/main.pl:1081]: the cli now requires you to pass in ids like so:
./terminusdb doc get admin/foo --ids '["Foo/1","Foo/2"]'

So as a json list.

Would probably be nicer if you could directly specify a comma separated list.

**Comment by matko** [comment]
> If I expect IDs to be returned, I want an error. If I ask exploratively I want what I can get. In the my current use case I would expect an error is the ID can't be found, as it implies that a new revision was likely created between the time of check and the time of update/request of my model.

Specifically on this point, note that we actually support data version headers which assert that the data you're querying is at the revision you expect it to be.

Every request that queries or updates the database will result in a `TerminusDB-Data-Version` header in the response. In any subsequent query/update request, this header can be resubmitted, and the operation will fail if the version mismatches.

**Pull request merged**